src/main/scala/chisel3/testing/scalatest/HasConfigMap.scala

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.testing.scalatest
+
+import org.scalatest.{ConfigMap, TestSuite, TestSuiteMixin}
+import scala.util.DynamicVariable
+
+/** A Scalatest test suite mix-in that provides access to command line options.
+  *
+  * This expose any keys/values passed using the `-D<key>=<value` command line
+  * option to the test via a `Map`.
+  *
+  * For example, you can invoke Scalatest passing the `foo=bar` option like so:
+  * {{{
+  * ./mill 'chisel[2.13.16].test.testOnly' 'fooTest' -Dfoo=bar
+  * }}}
+  *
+  * Inside your test, if the `configMap` member function is accessed this will
+  * return:
+  * {{{
+  * Map(foo -> bar)
+  * }}}
+  *
+  * This is intended to be a building block of more complicated tests that want
+  * to customize their execution via the command line.  It is advisable to use
+  * this sparingly as tests are generally not intended to change when you run
+  * them.
+  */
+trait HasConfigMap extends TestSuiteMixin { self: TestSuite =>
+
+  // Implement this via a `DynamicVariable` pattern that will be set via the
+  // `withFixture` method.  The `super.withFixture` function must be called to
+  // make this mix-in "stackable" with other mix-ins.
+  private val _configMap: DynamicVariable[Option[ConfigMap]] = new DynamicVariable[Option[ConfigMap]](None)
+  abstract override def withFixture(test: NoArgTest) = {
+    _configMap.withValue(Some(test.configMap)) {
+      super.withFixture(test)
+    }
+  }
+
+  /** Return the config map which contains all command line options passed to Scalatest.
+    *
+    * This is only valid during a test.  It will be `None` if used outside a
+    * test.
+    */
+  def configMap: Option[ConfigMap] = _configMap.value
```

Comment:
I understand the motivation for making it an `Option`, but I'm wondering if just returning `ConfigMap` and erroring if it isn't available might be more desirable. Is there ever a case that one would want the `None` over the exception? Maybe we provide both but make the default API error? e.g.

```suggestion
  /** Return the config map which contains all command line options passed to Scalatest.
    *
    * This is only valid during a test. It throw an exception if used outside a test.
    */
  def configMap: Option[ConfigMap] = configMapOpt.getOrElse {
    throw new Exception("ScalaTest configMap is only available within tests (e.g. it(\"works\") { <here> })")
  }

  /** Return the config map which contains all command line options passed to Scalatest.
    *
    * This is only valid during a test. It will be `None` if used outside a
    * test.
    */  
  def configMapOpt: Option[ConfigMap] = _configMap.value
```

Reply:
The logic was that it would be better to not have an exception. However, I think it's fine to only provide that. I _really_ don't want people to even be looking at this unless it's in a test. I'll change to _only_ provide the API which can throw.

Reply:
Switched to the assertion approach and added tests in: 904add2db